File: examples/swift-client/Sources/AppwriteModels/Country.swift

/// Country
public class Country {

    /// Country name.
    public let name: String

    /// Country two-character ISO 3166-1 alpha code.
    public let code: String

    init(
        name: String,
        code: String
    ) {
        self.name = name
        self.code = code
    }

    public static func from(map: [String: Any]) -> Country {
        return Country(
            name: map["name"] as! String,
            code: map["code"] as! String
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "name": name as Any,
            "code": code as Any
        ]
    }
            
}

File: examples/flutter/lib/src/models/country.dart
part of appwrite.models;

/// Country
class Country {
    /// Country name.
    final String name;
    /// Country two-character ISO 3166-1 alpha code.
    final String code;

    Country({
        required this.name,
        required this.code,
    });

    factory Country.fromMap(Map<String, dynamic> map) {
        return Country(
            name: map['name'],
            code: map['code'],
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "name": name,
            "code": code,
        };
    }
}
